### PR TITLE
Epic/payment epc qr code

### PR DESCRIPTION
**Pull request by julianzimmermann**

### CODE REVIEW

File: bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+ * @package FondOfOryx\Service\QrCodeGenerator
+ */
+class QrCodeGeneratorConfig extends AbstractBundleConfig
+{
+    /**
+     * @return string
+     */
+    public function getFormat(): string
+    {
+        return $this->get(QrCodeGeneratorConstants::QR_CODE_FORMAT, 'png');
```
Comment (daniel-rose) [inline bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php:20]: Use const for default config value.

File: bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * @return string
+     */
+    public function getEncoding(): string
+    {
+        return $this->get(QrCodeGeneratorConstants::QR_CODE_ENCODING, 'UTF-8');
```
Comment (daniel-rose) [inline bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php:28]: Use const for default config value.

File: bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * @return int
+     */
+    public function getErrorCorrectionLevel(): int
+    {
+        return $this->get(QrCodeGeneratorConstants::QR_CODE_ERROR_CORRECTION_LEVEL, 1);
```
Comment (daniel-rose) [inline bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php:36]: Use const for default config value.

File: bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * @return int
+     */
+    public function getSize(): int
+    {
+        return $this->get(QrCodeGeneratorConstants::QR_CODE_SIZE, 250);
```
Comment (daniel-rose) [inline bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php:44]: Use const for default config value.

File: bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * @return int
+     */
+    public function getMargin(): int
+    {
+        return $this->get(QrCodeGeneratorConstants::QR_CODE_MARGIN, 5);
```
Comment (daniel-rose) [inline bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php:52]: Use const for default config value.

File: bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * @return int
+     */
+    public function getRoundedBlockSizeMode(): int
+    {
+        return $this->get(QrCodeGeneratorConstants::QR_CODE_ROUNDED_BLOCK_SIZE_MODE, 1);
```
Comment (daniel-rose) [inline bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php:60]: Use const for default config value.

File: bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * @return array
+     */
+    public function getForegroundColor(): array
+    {
+        return $this->get(QrCodeGeneratorConstants::QR_CODE_FOREGROUND_COLOR, [0, 0, 0]);
```
Comment (daniel-rose) [inline bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php:68]: Use const for default config value.

File: bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    /**
+     * @return array
+     */
+    public function getBackgroundColor(): array
+    {
+        return $this->get(QrCodeGeneratorConstants::QR_CODE_BACKGROUND_COLOR, [255, 255, 255]);
```
Comment (daniel-rose) [inline bundles/qr-code-generator/src/FondOfOryx/Service/QrCodeGenerator/QrCodeGeneratorConfig.php:76]: Use const for default config value.

File: composer.json [hunk 9]
```diff
@@ -15,7 +15,9 @@
   "require": {
     "php": ">=7.3",
     "ext-json": "*",
+    "endroid/qr-code": "^4.1.0",
     "fond-of-spryker/country": "dev-spryker_upgrade || dev-feature/paypal-refund",
+    "fond-of-spryker/prepayment": "dev-spryker_upgrade",
```
Comment (daniel-rose) [inline composer.json:20]: Please add a ticket for moving prepayment from fond-of-spryker to mono repo.

File: bundles/payment-epc-qr-code/composer.json [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+    {
+      "name": "Julian Zimmermann",
+      "email": "info@julianzimmermann.dev"
+    }
+  ],
+  "require": {
+    "php": ">=7.3",
+    "fond-of-oryx/qr-code-generator": "^1.0.0",
+    "fond-of-spryker/prepayment": "dev-spryker_upgrade",
+    "spryker/oms-extension": "^1.1.0"
```
Comment (daniel-rose) [inline bundles/payment-epc-qr-code/composer.json:15]: If 1.1.0 is correct, please use this in main composer also.

**Review by daniel-rose — APPROVED** [review]

**Review by SpiderDan98 — APPROVED** [review]

**Pull request merged**